dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/channel/resolver/impl/netty/SipResolverUdpTransport.java

```diff
@@ -0,0 +1,664 @@
+                            + requestBuffer.toString(Charset.defaultCharset()) + " remote addr: "
+                            + _currentSocketAddress + " local addr: " + local);
+
+                ChannelFuture writeFuture = writeBufferAsDatagram(requestBuffer);
+
+                if (!writeFuture.isDone()) {
+                    _writeState = WRITE_STATE_WRITE_ACTIVE;
+                    writeFuture.addListener(f -> {
+                        if (f.isSuccess()) {
+                            writeComplete();
+                        } else {
+                            writeError((Exception) f.cause());
+                        }
+                    });
+                    break;
+                } else {
+                    // write completed so call complete from here
+                    writeComplete();
+                }
+            } else {
+                _writeState = WRITE_STATE_IDLE;
+                break;
+            }
+        }
+
+        if (c_logger.isTraceEntryExitEnabled())
+            c_logger.traceExit(this, "SipResolverUdpTransport: drainRequestQueue: exit _writeState: " + _writeState);
+    }
+
+    private class SipResolverUdpTransportHandler extends SimpleChannelInboundHandler<ByteBuf> {
+
+        public SipResolverUdpTransportHandler() {
+        }
+
+        @Override
+        protected void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
+            if (c_logger.isTraceEntryExitEnabled()) {
+                c_logger.traceEntry(this, "channelRead0 message length: " + msg.readableBytes());
+            }
+            messageRead(ctx, msg.retain());
+        }
+
+        @Override
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable e) throws Exception {
+            if (c_logger.isTraceEntryExitEnabled()) {
+                c_logger.traceEntry(this, "exceptionCaught: " + e.getMessage());
+            }
+            readError(ctx, e);
+            // tell the channel to close; the netty framework will be notified
+            ctx.close();
+        }
+
+        /** Called when a new connection is established */
+        @Override
+        public void channelActive(ChannelHandlerContext ctx) throws Exception {
+            if (c_logger.isTraceEntryExitEnabled()) {
+                c_logger.traceEntry(this, "channelActive local: " + ctx.channel().localAddress() + " remote: "
+                        + ctx.channel().remoteAddress());
+            }
+        }
+
+        @Override
+        public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+            if (c_logger.isTraceEntryExitEnabled()) {
+                c_logger.traceEntry(this, "channelInactive remote disconnected: " + ctx.channel().remoteAddress());
+            }
+            channel = null;
+        }
+    }
+
+    private final class SipMessageBufferDatagramDecoder extends MessageToMessageDecoder<DatagramPacket> {
```

Comment:
I'm guessing this class is this way to parallel how a class by the same name is used from GenericUDPChain.java

Reply:
There is a class with same prototype in netty transport that does slightly different things.   Not sure how this came to be but I'll leave it for now since they are different.